staging/src/k8s.io/kubectl/pkg/cmd/cmd.go

```diff
@@ -100,11 +100,13 @@ var defaultConfigFlags = genericclioptions.NewConfigFlags(true).WithDeprecatedPa
 
 // NewDefaultKubectlCommand creates the `kubectl` command with default arguments
 func NewDefaultKubectlCommand() *cobra.Command {
+	ioStreams := genericiooptions.IOStreams{In: os.Stdin, Out: os.Stdout, ErrOut: os.Stderr}
+	defaultConfigFlags = defaultConfigFlags.WithWarningPrinter(ioStreams)
```

Comment:
this is mutating a package variable in this constructor, which doesn't seem correct

Reply:
Fully agree about the concern in here.

I doubt that we get any benefit by defining this unexported default config as global variable. Only two places are using this and both of them are in same package(cmd.go). I'd prefer remove `defaultConfigFlags` altogether, what do you think?